WORKSPACE

```diff
@@ -8,13 +8,19 @@ load(
 http_archive(
     name = "io_bazel_rules_go",
     sha256 = "301c8b39b0808c49f98895faa6aa8c92cbd605ab5ad4b6a3a652da33a1a2ba2e",
-    urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.18.0/rules_go-0.18.0.tar.gz"],
+    urls = [
+        "https://github.com/bazelbuild/rules_go/releases/download/0.18.0/rules_go-0.18.0.tar.gz",
+        "https:/storage.googleapis.com/builddeps/301c8b39b0808c49f98895faa6aa8c92cbd605ab5ad4b6a3a652da33a1a2ba2e",
```

Comment:
does ordering matter here with the URLs? is there any way we can attempt to try our internal mirror first or something to guarantee all the artifacts in this pr match the corresponding shas? 

also, any thoughts about how we can make an automated process that generates this kind of change for us every time a dependency changes?

Reply:
@davidvossel as far as I know it will probe both and pick the faster, or the non-failing mirror. These files were generated by a tool which I want to evolve into exactly what you describe: https://github.com/kubevirt/project-infra/pull/226. The PR is a mess right now, and only the `upload` command works, which parsed the WORKSPACE, detected missing mirrored artifacts, uploaded them and dumped the changed WORKSPACE file. Checking the shasum on upload is planned. Right now bazel verifies it when downloading.

The final goal of https://github.com/kubevirt/project-infra/pull/226 is this:

 * A prow plugin, which puts a `hold/non-mirrored-dependency` label, if it finds a non-mirrored dependency.
 * Maintainers can write `/upload`
 * The plugin does the upload and pushes a commit to the PR (if that was forbidden by the creator it would post the diff and complain)
 * The label is removed
 * We can continue with /lgtm and /approve

Reply:
> Maintainers can write /upload

whoa, that's awesome!

Reply:
that's a great plan indeed! :+1: